openquake/hazardlib/source/base.py

```diff
@@ -40,6 +40,8 @@ class BaseSeismicSource(with_metaclass(abc.ABCMeta)):
     __slots__ = ['source_id', 'name', 'tectonic_region_type',
                  'trt_model_id', 'weight', 'seed', 'id']
 
+    MODIFICATIONS = abc.abstractproperty()
```

Comment:
If I'm not mistaken MODIFICATIONS is always an instance of a python set. Wouldn't it be easier to define it directly as an empty set in the base class?